examples/image_captioning/train.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+
+import chainer
+from chainer.datasets import TransformDataset
+from chainer import iterators
+from chainer import optimizers
+from chainer import training
+from chainer.training import extensions
+
+import datasets
+from model import ImageCaptionModel
+
+
+class TestModeEvaluator(extensions.Evaluator):
+
+    """Evaluates the model on the validation dataset."""
+
+    def evaluate(self):
+        with chainer.using_config('train', False), \
+                chainer.no_backprop_mode():
+            ret = super(TestModeEvaluator, self).evaluate()
+        return ret
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--out', type=str, default='result')
+    parser.add_argument('--mscoco-root', type=str, default='data')
+    parser.add_argument('--max-iters', type=int, default=50000)
+    parser.add_argument('--batch-size', type=int, default=128)
+    parser.add_argument('--dropout-ratio', type=float, default=0.5)
+    parser.add_argument('--val-n-imgs', type=int, default=1000)
+    parser.add_argument('--val-iter', type=int, default=100)
+    parser.add_argument('--log-iter', type=int, default=1)
+    parser.add_argument('--snapshot-iter', type=int, default=1000)
+    parser.add_argument('--rnn', type=str, default='lstm',
+                        choices=['lstm', 'nsteplstm'])
```

Comment:
How about using `nsteplstm` as default?

Reply:
Fixed for both the training and predicting scripts.